### PR TITLE
[WIP] script for distributed ssd training

### PR DESCRIPTION
**Pull request by eric-haibin-lin**
- added split sampler credit to @hetong007 
- added lr scheduler which can be used on dist kvstore server
- added non-blocking hybrid ssd multi box
- added an example script for ssd dist training

### CODE REVIEW

File: gluoncv/data/sampler.py [hunk 1]
```diff
@@ -0,0 +1,33 @@
...
+    part_index: int
+      The index of the part to read from
+    """
+    def __init__(self, length, num_parts=1, part_index=0):
+        # Compute the length of each partition
+        self.part_len = length // num_parts
+        # Compute the start index for this partition
+        self.start = self.part_len * part_index
+        # Compute the end index for this partition
+        self.end = self.start + self.part_len
```
Comment (eric-haibin-lin) [inline gluoncv/data/sampler.py]: Need to make sure self.end <= length

File: gluoncv/utils/lr_scheduler.py [hunk 2]
```diff
@@ -106,3 +111,109 @@ def update(self, i, epoch):
                     (1 + cos(pi * (T - self.warmup_N) / (self.N - self.warmup_N))) / 2
             else:
                 raise NotImplementedError
+
+class DistLRScheduler(lr_scheduler.LRScheduler):
```
Comment (hetong007) [inline gluoncv/utils/lr_scheduler.py]: Is there a way to make this PR and https://github.com/dmlc/gluon-cv/pull/353 compatible?

**Comment by hetong007** [comment]
Besides the scripts, we need a tutorial helping users to set up the environment. In my (limited) experience it is actually more frustrating than implementing the script.

**Comment by eric-haibin-lin** [comment]
@hetong007 I agree. I have a quip doc draft for step-by-step training setup but I do not yet have the cycle to polish it for wider audience

**Comment by hetong007** [comment]
Then maybe we can at least provide a bash file and log for distributed training of resnet50 on imagenet.

**Comment by zhreshold** [comment]
@eric-haibin-lin  Will this still be up to date after the unified Dist api modification? https://github.com/apache/incubator-mxnet/pull/17010

**Comment by eric-haibin-lin** [comment]
yes #17010 is backward compatible
